src/web_app/js/background.js

```diff
@@ -9,14 +9,152 @@
 /* More information about these options at jshint.com/docs/options */
 
 // Variables defined in and used from chrome.
-/* globals chrome */
+/* globals chrome, sendAsyncUrlRequest, Constants, parseJSON */
 
 'use strict';
-chrome.app.runtime.onLaunched.addListener(function() {
-  chrome.app.window.create('appwindow.html', {
-    'width': 800,
-    'height': 600,
-    'left': 0,
-    'top': 0
+(function() {
+  chrome.app.runtime.onLaunched.addListener(function() {
+    chrome.app.window.create('appwindow.html', {
+      'width': 800,
+      'height': 600,
+      'left': 0,
+      'top': 0
+    });
   });
-});
+
+  // Send event notification from background to window.
+  var sendWSEventMessageToWindow = function(port, wsEvent, data) {
+    var message = {
+        action: Constants.WS_ACTION,
+        wsAction: Constants.EVENT_ACTION,
+        wsEvent: wsEvent
+    };
+    if (data) {
+      message.data = data;
+    }
+    trace('B -> W: ' + JSON.stringify(message));
+    try {
+      port.postMessage(message);
+    } catch (ex) {
+      trace('Error sending message: ' + ex);
+    }
+  };
+
+  var handleWebSocketRequest = function(port, message) {
+    if (message.wsAction === Constants.WS_CREATE_ACTION) {
+      trace('RWS: creating web socket: ' + message.wssUrl);
+      var ws = new WebSocket(message.wssUrl);
+      port.wssPostUrl_ = message.wssPostUrl;
+      ws.onopen = function() {
+        trace('RWS: onopen');
```

Comment:
nit: the trace before sendWSEventMessageToWindow could probably be removed, since sendWSEventMessageToWindow already calls trace.